nucleus/deploy/client.py

```diff
@@ -0,0 +1,300 @@
+                imported list of packages.
+            env_params: A dictionary that dictates environment information e.g.
+                the use of pytorch or tensorflow, which cuda/cudnn versions to use.
+                Specifically, the dictionary should contain the following keys:
+                "framework_type": either "tensorflow" or "pytorch".
+                "pytorch_version": Version of pytorch, e.g. "1.5.1", "1.7.0", etc. Only applicable if framework_type is pytorch
+                "cuda_version": Version of cuda used, e.g. "11.0".
+                "cudnn_version" Version of cudnn used, e.g. "cudnn8-devel".
+                "tensorflow_version": Version of tensorflow, e.g. "2.3.0". Only applicable if framework_type is tensorflow
+            gpu_type: If specifying a non-zero number of gpus, this controls the type of gpu requested. Current options are
+                "nvidia-tesla-t4" for NVIDIA T4s, or "nvidia-tesla-v100" for NVIDIA V100s.
+
+        Returns:
+             A ModelEndpoint object that can be used to make requests to the endpoint.
+
+        """
+        if requirements is None:
+            requirements_inferred = find_packages_from_imports(globals())
+            requirements = [
+                f"{key}=={value}"
+                for key, value in requirements_inferred.items()
+            ]
+            logger.info(
+                "Using \n%s\n for model endpoint %s",
+                requirements,
+                service_name,
+            )
+            # TODO test
+        payload = dict(
+            service_name=service_name,
+            env_params=env_params,
+            bundle_name=model_bundle.name,
+            cpus=cpus,
+            memory=memory,
+            gpus=gpus,
+            gpu_type=gpu_type,
+            min_workers=min_workers,
+            max_workers=max_workers,
+            per_worker=per_worker,
+            requirements=requirements,
+        )
+        if gpus == 0:
+            del payload["gpu_type"]
+        elif gpus > 0 and gpu_type is None:
+            raise ValueError("If nonzero gpus, must provide gpu_type")
+        resp = self.connection.post(payload, ENDPOINT_PATH)
+        endpoint_creation_task_id = resp["data"][
+            "endpoint_id"
+        ]  # Serverside needs updating
+        logger.info(
+            "Endpoint creation task id is %s", endpoint_creation_task_id
+        )
+        return AsyncModelEndpoint(endpoint_id=service_name, client=self)
+
+    # Relatively small wrappers around http requests
+
+    def list_bundles(self) -> List[ModelBundle]:
+        """
+        Returns a list of model bundles that the user owns.
+        TODO this route doesn't exist serverside
+        """
+        # resp = self.connection.get("model_bundle")
+        raise NotImplementedError
+
+    def list_model_endpoints(self) -> List[AsyncModelEndpoint]:
```

Comment:
nit - use consistent name: model_endpoint vs endpoint, model_bundle vs bundle

perhaps we should just delete all "model" since it's kind of meaningless?